packages/api/portfolios/createPortfolio.ts

```diff
@@ -0,0 +1,88 @@
+import "reflect-metadata";
+import cors from "@middy/http-cors";
+import jsonBodyParser from "@middy/http-json-body-parser";
+import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
+import middy from "@middy/core";
+import validator from "@middy/validator";
+import xssSanitizer from "../portfolioDrafts/xssSanitizer";
+import { ApiGatewayEventParsed } from "../utils/eventHandlingTool";
+import { APIGatewayProxyResult, Context } from "aws-lambda";
+import { ApiSuccessResponse, SuccessStatusCode } from "../utils/response";
+import { CORS_CONFIGURATION } from "../utils/corsConfig";
+import { createConnection } from "../utils/database";
+import { errorHandlingMiddleware } from "../utils/errorHandlingMiddleware";
+import { IpCheckerMiddleware } from "../utils/ipLogging";
+import { IPortfolio, IPortfolioCreate } from "../../orm/entity/Portfolio";
+import { PortfolioRepository } from "../repository/PortfolioRepository";
+import { validateRequestShape } from "../utils/shapeValidator";
+import { wrapSchema } from "../utils/schemaWrapper";
+import schema = require("../models/internalSchema.json");
+
+/**
+ * Creates a new Portfolio
+ * A new portfolioId will be generated by the database.
+ *
+ * @param event - The POST request from API Gateway
+ */
+export async function baseHandler(
+  event: ApiGatewayEventParsed<IPortfolio>,
+  context?: Context
+): Promise<APIGatewayProxyResult> {
+  const setupResult = validateRequestShape<IPortfolio>(event);
+  const portfolioBody = setupResult.bodyObject;
+
+  // set up database connection
+  const connection = await createConnection();
+  const portfolioRepository = connection.getCustomRepository(PortfolioRepository);
+  let insertedPortfolio: IPortfolioCreate;
+
+  try {
+    const insertResult = await portfolioRepository.createPortfolio(portfolioBody);
+    const portfolioId = insertResult.identifiers[0].id;
+    // query for the newly inserted portfolio
+    insertedPortfolio = await portfolioRepository.findOneOrFail({
+      select: [
+        "id",
+        "name",
+        "csp",
+        "description",
+        "dodComponents",
+        "owner",
+        "portfolioManagers",
+        "createdAt",
+        "updatedAt",
+        "archivedAt",
+        "administrators",
+        "provisioningStatus",
+      ],
+      where: { id: portfolioId },
+    });
+
+    console.log("Inserted Portfolio: " + JSON.stringify(insertedPortfolio));
+  } finally {
+    connection.close();
+  }
+
+  return new ApiSuccessResponse<IPortfolioCreate>(insertedPortfolio, SuccessStatusCode.CREATED);
+}
+
+const portfolioSchema = {
+  additionalProperties: schema.Portfolio.additionalProperties,
+  type: schema.Portfolio.type,
+  required: schema.Portfolio.required,
+  properties: {
+    ...schema.PortfolioBase.properties,
+    ...schema.Portfolio.properties,
+    ...schema.PortfolioSummary.properties,
+    ...schema.PortfolioAccess.properties,
```

Comment:
While this doesn't have the `BaseObject` properties (e.g., createdAt), should we allow other properties to pass by the validator like properties related to applications, funding and possibly other things by spreading everything? 

The spec only allows the request body to have `csp`, `dodComponents`, `name`, `portfolioAdministrators`, `portfolioManagers`, and `description`. I don't think the `schema.PortfolioSummary.properties` would be needed in that case and the `schema.Portfolio.properties.applications` might not be wanted either. I might suggest to align with the spec a little closer.

```suggestion
    ...schema.PortfolioBase.properties,
    ...schema.PortfolioAccess.properties,
    description: schema.Portfolio.properties.description,
    owner: schema.Portfolio.properties.owner,
    portfolioManagers: schema.Portfolio.properties.portfolioManagers,
```

Reply:
Thanks for the critical look.  Will revisit this, especially seeing Zach's extensive example.

Reply:
This was improved in 620f1e7468095690dabfd867b7970dedc240456d